config/jupyterhub_config.py

```diff
@@ -4,24 +4,40 @@
 https://jupyterhub.readthedocs.io/en/latest/tutorial/getting-started/config-basics.html
 https://jupyterhub.readthedocs.io/en/stable/reference/config-reference.html
 """
+
 import os
 
+import nativeauthenticator
+
 from jupyterhub_config.custom_spawner import VirtualEnvSpawner
 
 c = get_config()
 
-# Set the authenticator class
-# TODO: Change the authenticator class to a secure one (e.g. GitHubOAuthenticator)
-c.JupyterHub.authenticator_class = 'jupyterhub.auth.DummyAuthenticator'
-c.Authenticator.allowed_users = {'spark_user', 'test_user1', 'test_user2'}
-c.DummyAuthenticator.password = os.environ['JUPYTERHUB_ADMIN_PASSWORD']
+# Set the authenticator class to nativeauthenticator
+# ref: https://native-authenticator.readthedocs.io/en/latest/quickstart.html
+c.JupyterHub.authenticator_class = 'native'
+c.JupyterHub.template_paths = [f"{os.path.dirname(nativeauthenticator.__file__)}/templates/"]
+c.NativeAuthenticator.open_signup = True
+c.NativeAuthenticator.check_common_password = True
+c.NativeAuthenticator.minimum_password_length = 6
 
+# Set up the admin user
 c.Authenticator.admin_users = {'spark_user'}
+# TODO set admin user password to os.environ['JUPYTERHUB_ADMIN_PASSWORD'] automatically - currently spark_user is created manually with the signup page
+c.Authenticator.allow_all = True
+
+c.JupyterHub.cookie_secret_file = f"{os.environ['JUPYTERHUB_USER_HOME']}/jupyterhub_cookie_secret"
+c.JupyterHub.db_url = f"sqlite:///{os.environ['JUPYTERHUB_USER_HOME']}/jupyterhub.sqlite"
 
 c.JupyterHub.spawner_class = VirtualEnvSpawner
 
+# Create a group to indicate users with read/write access to MinIO
+c.JupyterHub.load_groups = {
+    'minio_rw': [],
+}
```

Comment:
How do users get added to this group? I'm not sure how this actually works, come to think of it - the creds are in the env so the users by default have the creds right?

Reply:
you can do `'minio_rw': ['user_name1', 'user_name2'],` here or calling jupyterhub API to add users which is what I am planning to do. 

> users by default have the creds right

hmmm no. Users need to setup their own password when they signup an account.

Reply:
I mean the Minio creds. How would this group stop them from getting the Minio creds from the env?

Reply:
in next PR, I will set minio creds accordingly based on user group.

Reply:
yea. it will be set via by spawner `self.environment`.

Reply:
:+1: